server/gokbg3/grails-app/services/org/gokb/TitleAugmentService.groovy

```diff
@@ -203,12 +205,106 @@ class TitleAugmentService {
+            log.debug("No EZB result for ids of title ${titleInstance} (${titleInstance.ids.collect { it.value }})")
+            if (titleInstance.reviewRequests.collect {it.stdDesc == rr_info}.size() == 0) {
+              reviewRequestService.raise(
+                  titleInstance,
+                  "Check for reference ID",
+                  "No EZB matches for linked ISSNs",
+                  null,
+                  null,
+                  null,
+                  rr_info,
+                  editorialGroup
+              )
+            }
+          }
+        }
+        else {
+          log.debug("Multiple EZB-ID candidates for title ${titleInstance}")
+          def nameCandidates = []
+          ezbCandidates.each {
+            if (it.title == titleInstance.name) {
+              nameCandidates.add (it)
+            }
+          }
+          if (nameCandidates.size() == 1) {
+            // found 1 EZB match by name matching
+            ezbId = EzbAPIService.getJourId(nameCandidates[0])
+            def newOrExistingEzbId = componentLookupService.lookupOrCreateCanonicalIdentifier('ezb', ezbId)
+            log.debug("Adding new EZB-ID ${newOrExistingEzbId}")
+            new Combo(fromComponent: titleInstance, toComponent: newOrExistingEzbId, type: comboTypeId).save(flush: true, failOnError: true)
+          }
+          else if (nameCandidates.size() == 0) {
+            // found multiple matches by ID matching but 0 EZB match by name matching (very unlikely)
+            if (titleInstance.ids.collect { it.namespace.value == 'issn' || it.namespace.value == 'eissn' }) {
+              log.debug("Multiple EZB results for ID, but no EZB result for names of title ${titleInstance} (${titleInstance.ids.collect { it.value }})")
+              if (titleInstance.reviewRequests.collect {it.stdDesc == rr_info}.size() == 0) {
+                reviewRequestService.raise(
+                    titleInstance,
+                    "No action required.",
+                    "No EZB matches for title name",
+                    null,
+                    null,
+                    null,
+                    rr_info,
+                    editorialGroup
+                )
+              }
+            }
+          }
+          else {
+            reviewRequestService.raise(
+                titleInstance,
+                "No action required.",
+                "Multiple EZB-IDs found for ISSN and title name",
+                null,
+                null,
+                ([candidates: ezbCandidates] as JSON).toString(),
+                rr_info,
+                editorialGroup
+            )
+          }
+        }
+      }
+      else {
+        log.debug("Skipping title with existing RR ..")
+      }
+    }
+  }
+
   private void setNewTitleInfo(titleInstance, info) {
-    if (!titleInstance.publishedFrom && info.publishedFrom) {
+    if ((!titleInstance.hasProperty('publishedFrom') || !titleInstance.publishedFrom) && info.publishedFrom) {
```

Comment:
The property is defined in the generic base class `TitleInstance`, therefore this check shouldn't be necessary.